stravalib/client.py

```diff
@@ -715,38 +764,28 @@ def upload_activity(self, activity_file, data_type, name=None, description=None,
                 raise TypeError("Invalid type specified for activity_file: {0}".format(type(activity_file)))
 
         valid_data_types = ('fit', 'fit.gz', 'tcx', 'tcx.gz', 'gpx', 'gpx.gz')
-        if not data_type in valid_data_types:
-            raise ValueError("Invalid data type {0}. Possible values {1!r}".format(data_type, valid_data_types))
+        if data_type not in valid_data_types:
+            raise ValueError(f'Invalid data type {data_type}. Possible values {valid_data_types!r}')
 
         params = {'data_type': data_type}
         if name is not None:
             params['name'] = name
         if description is not None:
             params['description'] = description
         if activity_type is not None:
-            warnings.warn(
-                'The "activity_type" parameter is undocumented in the Strava API. Its use '
-                'may lead to unexpected behavior or errors in the future.',
-                FutureWarning,
-                stacklevel=2
-            )
             if not activity_type.lower() in [t.lower() for t in model.Activity.TYPES]:
-                raise ValueError("Invalid activity type: {0}.  Possible values: {1!r}".format(activity_type, model.Activity.TYPES))
-            params['activity_type'] = activity_type
+                raise ValueError(f'Invalid activity type: {activity_type}. Possible values: {model.Activity.TYPES!r}')
+            warn_param_unofficial('activity_type')
+            params['activity_type'] = activity_type.lower()
         if private is not None:
-            warnings.warn(
```

Comment:
same question here. just a general question so i understand!

Reply:
It's still deprecated, but I moved the actual warning into a separate function.